cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaClusterTest.java

```diff
@@ -1494,4 +1501,191 @@ public void testGetExternalAdvertisedUrlWithoutOverrides() {
         assertEquals("0://some-host.com:12345", kc.getExternalAdvertisedUrl(0, "some-host.com", "12345"));
         assertEquals("0://:12345", kc.getExternalAdvertisedUrl(0, "", "12345"));
     }
+
+    @Test
+    public void testGeneratePersistentVolumeClaims()    {
+
+        /**********
+         * Persistent storage with claim deletion
+         */
```

Comment:
I'm guilty of writing this kind of test myself, but I think it would be better to have a different test method for each of these cases. It makes the tests a lot clearer when future refactoring causes some of them to fail.

Reply:
Changed. Honestly, I was my self a bit undecided about this when writing the tests,